src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -1436,7 +1436,7 @@ public static IEnumerable<int> Range(int start, int count)
 
         private static IEnumerable<int> RangeIterator(int start, int count)
         {
-            for (int i = 0; i < count; i++) yield return start + i;
+            for (int copy = start; start < copy + count; start++) yield return start;
```

Comment:
The goal here is to avoid one addition per iteration?  I'd be surprised if this transformation achieved that, in particular given how the C# compiler rewrites the state machine for iterators today.  If that's the goal, you'd probably instead want to write it as:
```C#
for (int end = start + count; start < end; start++) 
    yield return start;
```